lizmap/modules/lizmap/classes/lizmapServices.class.php

```diff
@@ -178,11 +178,46 @@ class lizmapServices
     public $wmsMaxHeight = 3000;
 
     /**
-     * URL to the API exposed by the Lizmap plugin for QGIS Server.
+     * QGIS server JSON metadata file name.
      *
      * @var string
      */
-    public $lizmapPluginAPIURL = '';
+    public $qgisServerMetadata = '/server.json';
+
+    /**
+     * Custom URL to the API exposed by the Lizmap plugin for QGIS Server.
+     * Private property which can be an empty string if QGIS FCGI is used.
+     *
+     * @var string
+     */
+    private $lizmapPluginAPIURL = '';
+
+    /**
+     * Host URL to the Lizmap QGIS Server API, taking care of the QGIS Server context : FCGI, QJazz, etc.
+     *
+     * @var string
+     */
+    public function hostLizmapAPI()
+    {
+        if (empty($this->lizmapPluginAPIURL)) {
+            // When the Lizmap API URL is not set, we use the WMS server URL only
+            // and we add '/lizmap'/ at then end
+            return rtrim($this->wmsServerURL, '/').'/lizmap/';
+        }
+
+        // When the Lizmap API URL is set
+        return rtrim($this->lizmapPluginAPIURL, '/');
+    }
+
+    /**
+     * URL to the JSON QGIS Server metadata, taking care of the QGIS Server context : FCGI, QJazz, etc.
+     *
+     * @var string
+     */
+    public function urlLizmapQgisServerMetadata()
+    {
+        return $this->hostLizmapAPI.$this->qgisServerMetadata;
+    }
```

Comment:
Can you move these methods after `__construct` with the others getter and rename it with `get` suffix like `getHostLizmapApi` and `getQgisServerMetadataUrl` ? thanks